airbyte-db/lib/src/main/java/io/airbyte/db/factory/DataSourceFactory.java

```diff
@@ -0,0 +1,251 @@
+/*
+ * Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.db.factory;
+
+import com.zaxxer.hikari.HikariConfig;
+import com.zaxxer.hikari.HikariDataSource;
+import java.util.Map;
+import javax.sql.DataSource;
+
+/**
+ * Temporary factory class that provides convenience methods for creating a {@link DataSource}
+ * instance. This class will be removed once the project has been converted to leverage an
+ * application framework to manage the creation and injection of {@link DataSource} objects.
+ *
+ * This class replaces direct calls to {@link io.airbyte.db.Databases}.
+ */
+public class DataSourceFactory {
+
+  /**
+   * Constructs a new {@link DataSource} using the provided configuration.
+   *
+   * @param username The username of the database user.
+   * @param password The password of the database user.
+   * @param driverClassName The fully qualified name of the JDBC driver class.
+   * @param jdbcConnectionString The JDBC connection string.
+   * @return The configured {@link DataSource}.
+   */
+  public static DataSource create(final String username,
+                                  final String password,
+                                  final String driverClassName,
```

Comment:
Would it make sense to create an enum with all of the driver class name values that we support (i.e. the keys in the `JDBC_URL_FORMATS` below), and use that enum as the parameter type here? This way we ensure at compile time that these methods are only called with supported class names.

I guess this probably isn't a huge deal if we will be removing this class anyway once we switch to DI

Reply:
@lmossman Definitely.  I'm currently dealing with strings all over the place in the code as I do the refactor that uses these utilities.  I will introduce that change as part of that PR.